android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/ProjectControlsListAdapter.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * This file is part of BOINC.
+ * http://boinc.berkeley.edu
+ * Copyright (C) 2021 University of California
+ *
+ * BOINC is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * BOINC is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package edu.berkeley.boinc.adapter
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.TextView
+import androidx.appcompat.content.res.AppCompatResources
+import edu.berkeley.boinc.ProjectsFragment
+import edu.berkeley.boinc.ProjectsFragment.ProjectControl
+import edu.berkeley.boinc.R
+import edu.berkeley.boinc.rpc.RpcClient
+
+class ProjectControlsListAdapter(
+    context: Context?,
+    // ID of control texts in strings.xml
+    private val entries: List<ProjectControl>
+) : ArrayAdapter<ProjectControl>(
+    context!!, R.layout.projects_controls_listitem_layout, entries
+) {
```

Comment:
```suggestion
class ProjectControlsListAdapter(
    context: Context,
    // ID of control texts in strings.xml
    private val entries: List<ProjectControl>
) : ArrayAdapter<ProjectControl>(
    context, R.layout.projects_controls_listitem_layout, entries
) {
```

Reply:
We're passing nullable type here, so this code fails. I'll keep the original code

Reply:
Ok, I made a recheck and fixed it properly